### DISCUSSION TITLE
How to use alias in LEFT JOIN with alasql?

### DISCUSSION THREAD

**Discussion by arnemorken** [agershun/alasql#1932] [Q&A] [answered]
This simple query works fine in MySQL, but fails in AlaSQL - why?

```
SELECT DISTINCT any_event.*, temp.event_name AS parent_name
FROM any_event
LEFT JOIN any_event AS temp ON any_event.parent_id=temp.event_id
WHERE any_event.event_id=3

```
The error code from AlaSQL is (shortened for readability):

```
SELECT any_event.*, temp.event_name AS p
--------------------^
Expecting 'LITERAL', 'BRALITERAL', 'LPAR', [.....] 'MINUS', 'ATLBRA', 'LCUR', got 'TEMP'
```

Tried it with AlaSQL versions 4.3.1 and 3.0.0.

**Comment by arnemorken** [comment, marked as answer]
Appearantly, "temp" is a reserved keyword in AlaSQL (though it's not mentioned anywhere). When I switch "temp" with "tmp", everything works as expected. Go figure!

**Comment by mathiasrw** [comment]
Hmmm. You are right. I dont know why TEMP is reserved. I dont think its normal SQL syntax. I will avoid breaking compatibility and instead add it to the documentation. Thank you for letting the world know.
